src/imap/scan_folders.rs

```diff
@@ -101,6 +93,20 @@ impl Imap {
         last_scan.replace(Instant::now());
         Ok(true)
     }
+
+    /// Returns the names of all folders on the IMAP server that are not in `exclude`.
+    pub async fn list_folders(
+        self: &mut Imap,
+        context: &Context,
+    ) -> Result<Vec<async_imap::types::Name>> {
+        let session = self.session.as_mut();
+        let session = session.context("list_folders() IMAP No Connection established")?;
```

Comment:
I have started adding `context` everywhere recently and don't add the function name to context. Instead, function name could be added to the call of `list_folders`, e.g. `list_folders(...).await.context("failed to list folders")?` if needed. Most, if not all, `context` and `with_context` calls in DC core currently don't have function names, at least I have not found any from a quick (rip)grep.